src/Raven.Server/Documents/Commands/WaitForRaftCommands.cs

```diff
@@ -2,27 +2,30 @@
 using System.Net.Http;
 using Raven.Client.Http;
 using Raven.Client.Json;
+using Raven.Server.Monitoring.Snmp.Objects.Database;
+using Raven.Server.Utils;
 using Sparrow.Json;
 
 namespace Raven.Server.Documents.Commands
 {
     public class WaitForRaftCommands : RavenCommand
     {
         private readonly List<long> _indexes;
+        private readonly int? _shardNumber;
 
-        public WaitForRaftCommands(List<long> indexes)
+        public WaitForRaftCommands(List<long> indexes, int? shardNumber = null)
```

Comment:
If we need to do have it then it means that we are using wrong executor at some point